### ISSUE TITLE
1.1.162 seems to introduce some glitches in handling `ParamSpec`

### ISSUE THREAD

**Issue by MapleCCC** [microsoft/pyright#2196]
**To Reproduce**
```python
from collections.abc import Callable
from typing_extensions import ParamSpec

P = ParamSpec("P")

def repr_func_call(func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> str:

    arg_reprs = [repr(arg) for arg in args]
    arg_reprs.extend(k + "=" + repr(v) for k, v in kwargs.items())

    return func.__name__ + "(" + ", ".join(arg_reprs) + ")"
```

Pyright complains that `"object*" is not iterable`, and `"items" is not a known member of ParamSpec`. It seems Pyright treats `args` as object somehow, and treats `kwargs` as `ParamSpec`.

**Expected behavior**
There should be no error emitted for the above code.

**VS Code extension or command-line**
Pyright command-line 1.1.162

**Comment by erictraut** [comment]
Thanks for the bug report. Yes, I agree this is incorrect behavior. It's related to a change I introduced in the fix to this bug report: https://github.com/microsoft/pyright/issues/2178. You're using `P.args` and `P.kwargs` in a way that is perfectly legitimate but is not covered by any of our test cases. I'll work on a fix and use your sample to plug the hole in our tests.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by MapleCCC** [comment]
Dear Eric, thank you for your quick update. I tried your fix in bb7e0d4, and despite that this issue is fixed, there seems to be still some problems lurking. I opened a new issue (#2200) to track them.

**Comment by erictraut** [comment]
Thanks for reporting the other bug. It's unrelated to this one, but it's straightforward to fix too.

**Comment by MapleCCC** [comment]
I try the fix in bb7e0d4, and it is good. I will close this issue then.

**Comment by erictraut** [comment]
This is included in Pyright 1.1.163, which I just published. It will also be in the next release of Pylance.
